### PR TITLE
singletons demonstration

### PR DESCRIPTION
**Pull request by KrasimirRaikov**
Demonstrating eager and lazy singletons and their differences.

### CODE REVIEW

File: src/main/java/com/clouway/singletons/WorkerModule.java [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+import com.google.inject.Singleton;
+
+/**
+ * @author Krasimir Raikov(raikov.krasimir@gmail.com)
+ */
+public class WorkerModule extends AbstractModule {
+  @Override
+  protected void configure() {
+    bind(Worker.class).in(Singleton.class);
+    bind(EagerWorker.class).asEagerSingleton();
```
Comment (panayotkulchev) [inline src/main/java/com/clouway/singletons/WorkerModule.java]: Could you add example using `@Singleton @Provides` too

File: pom.xml [hunk 2]
```diff
@@ -53,6 +53,18 @@
 
     <!-- Set of utility classes and functions-->
     <dependency>
+      <groupId>com.google.inject</groupId>
+      <artifactId>guice</artifactId>
+      <version>4.1.0</version>
+    </dependency>
+
+    <dependency>
```
Comment (panayotkulchev) [inline pom.xml]: Is it needed for this task?

Comment (KrasimirRaikov) [inline pom.xml]: It is not, I'll remove it.

Comment (KrasimirRaikov) [inline pom.xml]: removed it.

File: src/main/java/com/clouway/singletons/WorkerModule.java (hunk 1 above)
Comment (KrasimirRaikov) [inline src/main/java/com/clouway/singletons/WorkerModule.java]: I have made the ```Worker``` class with ```@Provides``` method with ```@Singleton``` annotation.

**Comment by KrasimirRaikov** [comment]
Comments fixed, waiting review.

**Comment by panayotkulchev** [comment]
LGTM!

**Pull request merged**